#
# Patch to fix a parfait complaint about use of unitilized variables (see bug
# 24751636).  I have submitted a bug ticket to MIT (see 8501 Potential use of
# uninitialized variable in kdc/main.c) but in the meantime we will patch this.
#
# Patch source: in-house
#

--- a/src/kdc/main.c
+++ b/src/kdc/main.c
@@ -656,6 +656,9 @@ initialize_realms(krb5_context kcontext, int argc, char **argv)
         hierarchy[1] = KRB5_CONF_HOST_BASED_SERVICES;
         if (krb5_aprof_get_string_all(aprof, hierarchy, &hostbased))
             hostbased = 0;
+    } else {
+        fprintf(stderr, _("KDC cannot initialize. Cannot get profile.\n"));
+        exit(1);
     }
 
     if (default_udp_ports == 0) {
